cmd/vcfFilter/expression.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"github.com/vertgenlab/gonomics/vcf"
+	"log"
+	"strconv"
+	"strings"
+)
+
+type operator string
+
+const (
+	equal        operator = "="
+	notEqual     operator = "!="
+	greater      operator = ">"
+	less         operator = "<"
+	greaterEqual operator = ">="
+	lessEqual    operator = "<="
+	present      operator = ""
+)
+
+// TODO complex expression parsing
+// parseExpression parses a ';' delimited string to a slice of boolean functions to test a vcf record.
+func parseExpression(input string, header vcf.Header, isFormatElseInfo bool, includeMissingInfo bool) testingFuncs {
+	var answer testingFuncs
+	input = strings.Trim(input, "\"") // trim quotations from the ends
+	expSlice := strings.Split(input, ";")
+	for _, exp := range expSlice {
+		var op operator
+		var tagValuePair []string
+		var tag, value string
+		op = searchOp(exp)
+		if op != present {
+			tagValuePair = strings.Split(exp, string(op))
+		} else {
+			tagValuePair = []string{exp}
+		}
+		tag = strings.Trim(tagValuePair[0], " ")
+		if len(tagValuePair) == 2 {
+			value = strings.Trim(tagValuePair[1], " ")
+		}
+		answer = append(answer, getRelationshipTest(tag, value, op, header, isFormatElseInfo, includeMissingInfo))
+	}
+	return answer
+}
+
+func searchOp(exp string) operator {
```

Comment:
Wondering if these functions should be housed in another package. Logical operations might be broadly applicable for other cmds

Reply:
Particularly with the TestInteger-like functions

Reply:
At first this was going to be in a package, but I ended up dropping a lot of planned features mostly relating to complex logic (AND/OR). I decided to put the dumbed-down version in the cmd until I had the time to make a full version. But, I an open to moving this to a package as-is if people feel strongly.